### ISSUE TITLE
no-duplicate-case incorrect error

### ISSUE THREAD

**Issue by aaronz8** [eslint/eslint#2038]
the second and 3rd cases both get a Duplicate case label error. They are not the same and their values are not the same.

switch(action.type) {

        case ActionTypes.CLICK_PROJECT:
          _currentID = action.projectID;
          this.emitChange();
          break;

        case ActionTypes.CREATE_MESSAGE:
          var message = ChatMessageUtils.getCreatedMessageData(
            action.text,
            action.currentThreadID
          );
          _projects[message.id] = message;
          this.emitChange();
          break;

        case ActionTypes.RECEIVE_RAW_PROJECTS:
          _addProjects(action.rawProjects);
          this.emitChange();
          break;

        default:
          // do nothing
      }

  34:8   error    Duplicate case label                           no-duplicate-case
  43:8   error    Duplicate case label                           no-duplicate-case

**Comment by lencioni** [comment]
Here is a simplified repro case:

```js
var condition = 1;
var cases = {
  a: 1,
  b: 2,
};

switch (condition) {
  case cases.a:
    break;
  case cases.b:
    break;
}
```

**Comment by doberkofler** [comment]
@lencioni @aaronz8 Thank for notifying. I've already submitted a pull request this morning (GMT) that fixes the problem.

**Comment by aaronz8** [comment]
@doberkofler Thanks for the quick response and fix!

**Comment by jokeyrhyme** [comment]
Haha! I was just about to file this. Thanks for fixing this!
